src/main/java/com/amazon/opendistroforelasticsearch/security/configuration/ConfigurationRepository.java

```diff
@@ -250,6 +226,33 @@ private boolean createSecurityIndexIfAbsent() {
         }
     }
 
+    private void waitForSecurityIndexToBeAtLeastYellow() {
+        LOGGER.info("Node started, try to initialize it. Wait for at least yellow cluster state....");
+        ClusterHealthResponse response = null;
+        try {
+            response = client.admin().cluster().health(new ClusterHealthRequest(opendistrosecurityIndex)
+                    .waitForActiveShards(1)
+                    .waitForYellowStatus()).actionGet();
+        } catch (Exception e1) {
+            LOGGER.debug("Caught a {} but we just try again ...", e1.toString());
+        }
+
+        while(response == null || response.isTimedOut() || response.getStatus() == ClusterHealthStatus.RED) {
```

Comment:
This loop goes on indefinite if the index is always red.

Is there any concern if the trying count being capped at a certain number?

Reply:
The current behaviour too is the same (except the infinite loop is after attempting to load the data into the index).
I don't think we should cap the loop as the plugin will not behave as intended if the data is not loaded into index or if the index is red.